nav2_map_server/test/unit/test_occ_grid.cpp

```diff
@@ -33,37 +33,51 @@
 #include <experimental/filesystem>
 #include <stdexcept>
 #include <string>
+#include <memory>
 
-#include "nav2_map_server/map_representations/occ_grid_server.hpp"
+#include "nav2_map_server/occ_grid_loader.hpp"
 #include "test_constants/test_constants.h"
 
 #define TEST_DIR TEST_DIRECTORY
 
 using namespace std; // NOLINT
 using std::experimental::filesystem::path;
 
-class OccGridTest : public nav2_map_server::OccGridServer
+class RclCppFixture
 {
 public:
-  OccGridTest() {}
-  ~OccGridTest() {}
+  RclCppFixture() {rclcpp::init(0, nullptr);}
+  ~RclCppFixture() {rclcpp::shutdown();}
+};
+
+RclCppFixture g_rclcppfixture;
+
+class TestMapLoader : public nav2_map_server::OccGridLoader
+{
+public:
+  explicit TestMapLoader(rclcpp::Node * node)
+  : OccGridLoader(node)
+  {
+  }
 
-  nav_msgs::srv::GetMap::Response GetMap() {return occ_resp_;}
+  nav_msgs::msg::OccupancyGrid getOccupancyGrid()
+  {
+    return msg_;
+  }
 };
 
-class MapServerTest : public ::testing::Test
+class MapLoaderTest : public ::testing::Test
 {
 public:
-  MapServerTest()
+  MapLoaderTest()
   {
-    OccTest = new OccGridTest;
-    auto test_yaml = path(TEST_DIR) / path(g_valid_yaml_file);
-    OccTest->LoadMapInfoFromFile(test_yaml.string());
+    node_ = std::make_shared<rclcpp::Node>("MapLoaderTest");
+    map_loader_ = new TestMapLoader(node_.get());
   }
 
 protected:
-  OccGridTest * OccTest;
-  nav_msgs::srv::GetMap::Response map_resp;
+  rclcpp::Node::SharedPtr node_;
```

Comment:
Here you added a shared ptr, but in the other file you had a raw pointer

Reply:
Yeah, I (we) are not consistent in the code. There are some places where a raw node pointer is used and others where the SharedPtr is used. Sometimes a SharedPtr is needed and we're in the constructor where we only have a this pointer. There are various situations and we need to scrub all of this to make it consistent.